### ISSUE TITLE
Listbox Widget A11y Issues

### ISSUE THREAD

**Issue by tannerhuynh** [reach/reach-ui#928]
### Current Behavior

[Listbox](https://reach.tech/listbox/) widget that triggers dropdown has role='button' and has no aria-expanded attribute. Options do not have aria-selected attribute.

### Expected behavior

Listbox should have role of combox with aria-expanded attribute based on whether list is expanded or not. Options should have attributes for aria-selected.

### Reproducible example

Current usage of listbox.

### Suggested solution(s)

https://w3c.github.io/aria-practices/examples/combobox/combobox-select-only.html

We had a third party wcag experts review the component and they flagged it as not entirely understandable. The component has role="button" and aria-haspopup="listbox" attributes, which cause the control to be announced as a menu instead of a select or combobox control. Assistive technology users may not understand how to interact with this control.

**Comment by danielsimao** [comment]
@tannerhuynh

Meanwhile you are waiting for a fix. You can solve part of your issue by doing the following:

```
<ListboxInput>
  {({ value, valueLabel, isExpanded }) => (
    <ListboxButton aria-expanded={isExpanded}>
      <span data-value={value}>{valueLabel}</span>
    </ListboxButton>
    <ListboxPopover>
      <ListboxList>
        <ListboxOption value="apple">Apple 🍏</ListboxOption>
        <ListboxOption value="orange">Orange 🍊</ListboxOption>
        <ListboxOption value="banana">Banana 🍌</ListboxOption>
      </ListboxList>
    </ListboxPopover>
  )}
</ListboxInput>
```

**Comment by danielsimao** [comment]
Its seems to only have the aria-expanded when open. @tannerhuynh you would like to have it as well when close, meaning: `aria-expanded=false`?

**Comment by danielsimao** [comment]
I don't think its a problem. The guys in reachui team followed w3 documentation. https://www.w3.org/TR/2017/NOTE-wai-aria-practices-1.1-20171214/examples/listbox/listbox-collapsible.html
